controllers/remotesecret_controller.go

```diff
@@ -156,11 +168,54 @@ func (r *RemoteSecretReconciler) Reconcile(ctx context.Context, req reconcile.Re
 		return ctrl.Result{}, nil
 	}
 
-	if remoteSecret.Spec.Target.Namespace != "" {
-		err = r.deployToNamespace(ctx, remoteSecret, secretData)
+	aerr := &AggregatedError{}
+	r.processTargets(ctx, remoteSecret, secretData, aerr)
+
+	var returnedError error
+
+	if aerr.HasErrors() {
+		if serr := r.saveFailureStatus(ctx, remoteSecret, aerr); serr != nil {
+			lg.Error(serr, "failed to update the status with errors from dependent objects deploymet", "deploymentErrors", aerr)
+		}
+		returnedError = aerr
+	} else {
+		meta.SetStatusCondition(&remoteSecret.Status.Conditions, metav1.Condition{
+			Type:   string(api.RemoteSecretConditionTypeDeployed),
+			Status: metav1.ConditionTrue,
+			Reason: string(api.RemoteSecretReasonInjected),
+		})
+		if err = r.saveSuccessStatus(ctx, remoteSecret); err != nil {
+			lg.Error(err, "failed to update the status of the remote secret with success after deploying everything successfully")
+			returnedError = err
+		}
+	}
+
+	return ctrl.Result{}, returnedError
+}
+
+func (r *RemoteSecretReconciler) processTargets(ctx context.Context, remoteSecret *api.RemoteSecret, secretData *remotesecretstorage.SecretData, errorAggregate *AggregatedError) {
+	namespaceClassification := remotesecrets.ClassifyTargetNamespaces(remoteSecret)
```

Comment:
How this indexing work? Is order guaranteed in k8s objects ?

Reply:
it is explained in the doc of the `ClassifyTargetNamespaces` function.